xmppserver/src/main/java/org/jivesoftware/openfire/http/HttpBindServlet.java

```diff
@@ -340,7 +340,9 @@ protected static String getRemoteAddress(AsyncContext context)
 
         @Override
         public void onDataAvailable() throws IOException {
-            Log.trace("Data is available to be read from [" + remoteAddress + "]");
+            if( Log.isTraceEnabled() ) {
+                Log.trace("Data is available to be read from [" + remoteAddress + "]");
+            }
```

Comment:
For future reference, these three lines (and similar) can be done in one using SLF4J placeholders; 
`Log.trace("Data is available to be read from [{}]", remoteAddress);`
Won't ask you to change the PR on this basis, but just something to keep in mind for the future.